### PR TITLE
Feature: Custom environment variables

### PR DESCRIPTION
**Pull request by fatal10110**
Implements #1088

### CODE REVIEW

**Comment by pouledodue** [comment]
is possible to merge this, I need to set `keep_trailing_newline=True` in jinja

File: cookiecutter/environment.py [hunk 1]
```diff
@@ -7,6 +7,31 @@
 from .exceptions import UnknownExtension
 
 
+class CustomEnvironmentArgsMixin(object):
```
Comment (quantumlicht) [inline cookiecutter/environment.py]: While this is convenient, it seems like the cookiecutter prefers users using jinja2 extensions to extend functionality.

You can write a simple extension like this for your use case

```
class CustomDelimitersExtension(Extension):
    def __init__(self, environment):

        super(CustomDelimitersExtension, self).__init__(environment)
        environment.variable_start_string = "[["
        environment.variable_end_string = "]]"
```

Comment (quantumlicht) [reply cookiecutter/environment.py]: I would probably remove the addition of the mixin from this PR to make it easier to accept.
I think the key point you covered, is passing the env variable to `find_template`. Extensions are loaded when env is instantiated, so your variable delimiters would be taken into account.

**Comment by quantumlicht** [comment]
Can somebody from the maintainer team take a look at this? With my recommendation (to remove the new mixin), I think this serves as a bug fix to allow using custom delimiters properly.

**Comment by JustinPlute** [comment]
Any update on this?

**Comment by con-f-use** [comment]
See #1339 for an alternative implementation.

**Comment by stefan-caraiman** [comment]
bump

**Comment by kurtmckee** [comment]
I'm closing this PR due to its age.

**Pull request closed without merging**